tests/testdata/config/config-yml-invalid.yml

```diff
@@ -0,0 +1,2 @@
+process_count: 5
+invalid_yml
```

Comment:
Can you try to get around this two line file? I think as an alternative you could just mock this by writing an arbitrary string to the test temp directory and just use this file path for creating the configuration. I write a comment at the corresponding test as well suggesting an implementation.